apps/examples/fs_helper/fs_rw_test.c

```diff
@@ -0,0 +1,221 @@
+	} else {
+		printf("open fail please select O_RDONLY or O_WRONLY option\n");
+		return -1;
+	}
+
+	t = get_time();
+	fd = open(path, oflags);
+	if (fd == -1) {
+		printf("%s errno: %d\n", __FUNCTION__, errno);
+		exit(EXIT_FAILURE);
+	}
+	dt = get_time() - t;
+	total_time += dt;
+	rw_testlog(printopt, "open(%c):%dms, ", mode, dt);
+	return fd;
+}
+
+static void t_write(int fd, const void *buf, size_t nbytes, char *printopt)
+{
+	long t = 0;
+	long dt = 0;
+	int ret = 0;
+
+	t = get_time();
+	ret = write(fd, buf, nbytes);
+	if (ret == -1) {
+		printf("%s errno: %d\n", __FUNCTION__, errno);
+		exit(EXIT_FAILURE);
+	}
+	dt = get_time() - t;
+	total_time += dt;
+	rw_testlog(printopt, "write:%dms, ", dt);
+}
+
+static void t_close(int fd, char *printopt)
+{
+	long t = 0;
+	long dt = 0;
+	int ret = 0;
+
+	t = get_time();
+	ret = close(fd);
+	if (ret == -1) {
+		printf("%s errno: %d\n", __FUNCTION__, errno);
+		exit(EXIT_FAILURE);
+	}
+	dt = get_time() - t;
+	total_time += dt;
+	rw_testlog(printopt, "close:%dms\n", dt);
+}
+
+static void t_read(int fd, char *buf, size_t nbytes, char *printopt)
+{
+	long t = 0;
+	long dt = 0;
+	int ret = 0;
+
+	t = get_time();		
+	ret = read(fd, buf, nbytes);
+	if (ret == -1) {
+		printf("%s errno: %d\n", __FUNCTION__, errno);
+		exit(EXIT_FAILURE);
+	}
+	dt = get_time() - t;
+	total_time += dt;
+	itoa((int)buf[0], buf, 10);
+	rw_testlog(printopt, "read:%dms, data:%s ", dt, buf);
+}
+
+static void rw_operation(char *path, int size, int operation, int iteration, char *printopt)
+{
+	int fd = -1;
+	char data[size];
+	char buf[size];
+
+	data[0] = iteration;
+	rw_testlog(printopt, "==========iteration %d==========\n", iteration);
+	switch (operation) {
+	case 1:
+		fd = t_open(path, O_WRONLY | O_CREAT | O_TRUNC, printopt);
+		t_write(fd, data, sizeof(data), printopt);
+		t_close(fd, printopt);
+		fd = t_open(path, O_RDONLY, printopt);
+		t_read(fd, buf, sizeof(buf), printopt);
+		break;
+	case 2:
+		fd = t_open(path, O_RDONLY, printopt);
+		t_read(fd, buf, sizeof(buf), printopt);
+		break;
+	case 3:
+		fd = t_open(path, O_WRONLY | O_CREAT | O_TRUNC, printopt);
+		t_write(fd, data, sizeof(data), printopt);
+		break;
+	default:
+		return;
+	}
+	t_close(fd, printopt);
+	
+	rw_testlog(printopt, "================================\n");
+}
+
+void f_rw_performance(char *path, int size, int iteration, int operation, char *printopt)
+{
+	for (int i = 0; i < iteration; i++) {
+		rw_operation(path, size, operation, i, printopt);
+	}
+	printf("run_time: %d ms\n", total_time);
+}
+
+#ifdef CONFIG_BUILD_KERNEL
+int main(int argc, FAR char *argv[])
+#else
+int fs_rw_test_main(int argc, char *argv[])
+#endif
+{
+	int size;
+	int iteration;
+	int operation;
+
+	printf("FS R/W TEST!\n");
+	total_time = 0;
+	if (argc <= 5) {
+		printf("%s", USAGE);
+		return 0;
+	}
+
+	size = atoi(argv[2]);
+	iteration = atoi(argv[3]);
+	operation = atoi(argv[4]);
+
+	if (operation < 0 || operation > 3) {
+		printf("%s", USAGE);
+		return 0;
+	} else {
```

Comment:
You can remove else because there is return inside if operation.

Reply:
modified